src/log/format.cr

```diff
@@ -90,7 +92,12 @@ class Log
     # Parameters `before` and `after` can be provided to be written around
     # the value.
     # ```
-    # source(before: '[', after: ']') # => [http.server]
+    # Log.define_formatter TestFormatter, "#{source(before: '[', after: "] ")}#{message}"
+    #
+    # io = IO::Memory.new
+    # entry = Log::Entry.new("source", :info, "message", Log::Metadata.empty, nil)
+    # TestFormatter.format(entry, io)
+    # io.rewind.gets # => "[source] message"
     # ```
```

Comment:
I believe this is a minimum working code using `source`.
But the original pseudo code might be better due to this complexity. Thought?

Reply:
I like the `Log.define_formatter` part.

I'm between leaving just that or repeating a full setup as in the beginning of the `StaticFormatter`

```crystal
Log.setup(:info, Log::IOBackend.new(formatter: TestFormatter))
Log.for("foo.bar").info { "Hello" }    # => - [foo.bar] Hello
```